src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -29,6 +30,12 @@ class AutowirePass implements CompilerPassInterface
     private $definedTypes = array();
     private $types;
     private $ambiguousServiceTypes = array();
+    private $reflectionHelper;
+
+    public function __construct(ReflectionHelper $reflectionHelper = null)
```

Comment:
Instead of injecting an object, I suggest a string:
`$reflectionClassClassname = \ReflectionClass::class`
Could it be made to work this way?